### PR TITLE
[NO GBP]Grave Patches

### LINKED ISSUE #75553: You cannot refill graves anymore
Reporting client version: 514.1589

## Round ID:
[206518](https://scrubby.melonmesa.com/round/206518)

## Reproduction:
Dig up a grave, try to fill it any way you can tink.

### PR DESCRIPTION
**Pull request by SyncIt21**
## About The Pull Request

1. Fixes #75553 You can put stuff in the grave normally by clicking that item on the grave rather than throwing them in, this also means you can destroy graves by attacking them with items or just dig the grave with a shovel.
2. Graves can be closed again but "only" with a shovel and not instantly by hand which was the previous behaviour (which was dumb and nonsensical),
3. Add's examines & screentips for graves

## Changelog
:cl:
fix: graves can be closed again, but only with a shovel and not by hand.
qol: you can put items in graves normally without throwing them in
qol: examines & screentips for graves
/:cl:

### CODE REVIEW

File: code/modules/mapfluff/ruins/lavalandruin_code/elephantgraveyard.dm [hunk 1]
```diff
@@ -197,37 +193,71 @@
...
+		return ..()
+
+	//player is attempting to open/close the grave with a shovel
+	if(!user.combat_mode)
+		user.visible_message(
+			span_notice("[user] Is attempting to [opened ? "close" : "dig open"] [src]."),
+			span_notice("You start [opened ? "closing" : "digging open"] [src]."),
+		)
+		if(!weapon.use_tool(src, user, delay = 15, volume = 40))
+			return 1
```
Comment (san7890) [inline code/modules/mapfluff/ruins/lavalandruin_code/elephantgraveyard.dm:231]:
```suggestion
			return TRUE
```

File: code/modules/mapfluff/ruins/lavalandruin_code/elephantgraveyard.dm [hunk 2]
```diff
@@ -197,37 +193,71 @@
...
+			dug_closed = TRUE
+			close(user)
+		else if(open(user, force = TRUE))
+			user.add_mood_event("graverobbing", /datum/mood_event/graverobbing)
+			if(lead_tomb && first_open)
+				user.gain_trauma(/datum/brain_trauma/magic/stalker)
+				to_chat(user, span_boldwarning("Oh no, no no no, THEY'RE EVERYWHERE! EVERY ONE OF THEM IS EVERYWHERE!"))
+				first_open = FALSE
+
+		return 1
```
Comment (san7890) [inline code/modules/mapfluff/ruins/lavalandruin_code/elephantgraveyard.dm:243]:
```suggestion
		return TRUE
```

File: code/modules/mapfluff/ruins/lavalandruin_code/elephantgraveyard.dm [hunk 3]
```diff
@@ -197,37 +193,71 @@
...
-						to_chat(user, span_boldwarning("Oh no, no no no, THEY'RE EVERYWHERE! EVERY ONE OF THEM IS EVERYWHERE!"))
-						first_open = FALSE
-					return 1
-				return 1
-			else
-				to_chat(user, span_notice("You can't dig up a grave with \the [S.name]."))
-				return 1
-		else
-			to_chat(user, span_notice("The grave has already been dug up."))
 			return 1
```
Comment (san7890) [inline code/modules/mapfluff/ruins/lavalandruin_code/elephantgraveyard.dm:248]:
```suggestion
			return TRUE
```

File: code/modules/mapfluff/ruins/lavalandruin_code/elephantgraveyard.dm [hunk 4]
```diff
@@ -197,37 +193,71 @@
...
-				user.add_mood_event("graverobbing", /datum/mood_event/graverobbing)
-				deconstruct(TRUE)
-				return 1
-	return
+		user.visible_message(
+			span_notice("[user] Is attempting to remove [src]."),
+			span_notice("You start removing [src]."),
+		)
+		if(!weapon.use_tool(src, user, delay = 15, volume = 40) || !opened)
+			return 1
```
Comment (san7890) [inline code/modules/mapfluff/ruins/lavalandruin_code/elephantgraveyard.dm:255]:
```suggestion
			return TRUE
```

File: code/modules/mapfluff/ruins/lavalandruin_code/elephantgraveyard.dm [hunk 5]
```diff
@@ -197,37 +193,71 @@
...
+
+		to_chat(user, span_notice("You remove \the [src]  completely."))
+		user.add_mood_event("graverobbing", /datum/mood_event/graverobbing)
+		deconstruct(TRUE)
+		return 1
```
Comment (san7890) [inline code/modules/mapfluff/ruins/lavalandruin_code/elephantgraveyard.dm:260]:
```suggestion
		return TRUE
```

File: code/modules/mapfluff/ruins/lavalandruin_code/elephantgraveyard.dm [hunk 6]
```diff
@@ -197,37 +193,71 @@
...
-				return 1
-	return
+		user.visible_message(
+			span_notice("[user] Is attempting to remove [src]."),
+			span_notice("You start removing [src]."),
+		)
+		if(!weapon.use_tool(src, user, delay = 15, volume = 40) || !opened)
+			return 1
+
+		to_chat(user, span_notice("You remove \the [src]  completely."))
```
Comment (san7890) [inline code/modules/mapfluff/ruins/lavalandruin_code/elephantgraveyard.dm:257]:
```suggestion
		to_chat(user, span_notice("You remove \the [src] completely."))
```

spacing is wonky here

File: code/modules/mapfluff/ruins/lavalandruin_code/elephantgraveyard.dm [hunk 7]
```diff
@@ -138,32 +138,28 @@
...
-	cutting_tool = /obj/item/shovel
 	can_install_electronics = FALSE
+	can_weld_shut = FALSE
+	cutting_tool = null
 	paint_jobs = null
 
 	var/lead_tomb = FALSE
 	var/first_open = FALSE
-	var/grave_dug_open = FALSE
+	var/dug_closed = FALSE
```
Comment (san7890) [inline code/modules/mapfluff/ruins/lavalandruin_code/elephantgraveyard.dm:148]: would you mind dmdoccing all of these while you're here?

**Review by san7890 — CHANGES_REQUESTED** [review]
those are all meant to be booleans they were just done wrong

**Comment by SyncIt21** [comment]
Done

**Comment by SyncIt21** [comment]
@san7890 ready for merge?

**Pull request merged**